### PR TITLE
terraform: add Amazon Linux for libvirtd provider

### PR DESCRIPTION
**Pull request by gila**

### CODE REVIEW

File: terraform/aws-linux/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+## AWS Linux
+
+EKS uses this image as the base operating system. It is based on Redhat.  During
```
Comment (jonathan-teh) [inline terraform/aws-linux/README.md:3]: Red Hat Enterprise Linux 7.

File: terraform/aws-linux/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+## Usage
+
+Make sure you edit main.tf to set the proper variables like user keys, image
+etc.
+
+By default, a user "ec2-user" is available on the ISO, the password of this user
+is set within the meta-data file. Additionally, a new user is created that
+points to your local user and pick up the right SSH keys from there.
+
+Also, i'd recommend to the QCOW image locally as by default it will be fetched
```
Comment (jonathan-teh) [inline terraform/aws-linux/README.md:23]: to _download_ the QCOW...?

File: terraform/aws-linux/main.tf [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+variable "hostname_formatter" {
+	default = "amz-%d"
+}
+
+variable "ssh_user" {
+	default = "gila"
+}
+
+variable "ssh_key" {
+	default = "/home/gila/.ssh/id_rsa.pub"
```
Comment (jonathan-teh) [inline terraform/aws-linux/main.tf:14]: Similar to previous comment, `var.ssh_key`.

File: terraform/aws-linux/main.tf [hunk 4]
```diff
@@ -0,0 +1,137 @@
+variable "num_nodes" {
+	default = 1
+}
+
+variable "hostname_formatter" {
+	default = "amz-%d"
+}
+
+variable "ssh_user" {
+	default = "gila"
```
Comment (jonathan-teh) [inline terraform/aws-linux/main.tf:10]: Should this be `var.ssh_user` to pick up the one defined in the top-level variables.tf?

File: terraform/aws-linux/main.tf [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+variable "ssh_key" {
+	default = "/home/gila/.ssh/id_rsa.pub"
+
+}
+
+variable "private_key_path" {
+	default = "/home/gila/.ssh/id_rsa"
```
Comment (jonathan-teh) [inline terraform/aws-linux/main.tf:19]: Ditto, `var.private_key_path`.

File: terraform/aws-linux/main.tf [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+# a pool is where we store all images and cloud-init iso's
+resource "libvirt_pool" "aws-pool" {
+  name = "aws-pool"
+  type = "dir"
+  path = "/images"
+}
+
+# wget https://cdn.amazonlinux.com/os-images/2.0.20200602.0/kvm/amzn2-kvm-2.0.20200602.0-x86_64.xfs.gpt.qcow2
```
Comment (jonathan-teh) [inline terraform/aws-linux/main.tf:43]: Perhaps move the comment to the image_source section and point to https://cdn.amazonlinux.com/os-images/latest/kvm/ as the URL to get the latest image?

File: terraform/aws-linux/main.tf (hunk 4 above)
Comment (gila) [reply terraform/aws-linux/main.tf:10]: No, it does not use the module system at all. Its "stand-alone"

File: terraform/aws-linux/README.md (hunk 1 above)
Comment (gila) [reply terraform/aws-linux/README.md]: thx fixed

File: terraform/aws-linux/main.tf (hunk 3 above)
Comment (gila) [reply terraform/aws-linux/main.tf:14]: thx fixed.

**Pull request merged**